tests/functional/tier4/test_compile_linux_kernel_with_self_heal.py

```diff
@@ -0,0 +1,222 @@
+#  any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License along
+#  with this program; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+from random import choice
+
+from glusto.core import Glusto as g
+from glustolibs.gluster.gluster_base_class import GlusterBaseClass, runs_on
+from glustolibs.gluster.exceptions import ExecutionError
+from glustolibs.gluster.brick_libs import (bring_bricks_offline,
+                                           bring_bricks_online,
+                                           are_bricks_offline,
+                                           are_bricks_online, get_all_bricks)
+from glustolibs.gluster.glusterdir import mkdir
+from glustolibs.gluster.heal_libs import monitor_heal_completion
+from glustolibs.gluster.lib_utils import collect_bricks_arequal
+from glustolibs.gluster.volume_libs import get_subvols
+from glustolibs.io.utils import collect_mounts_arequal, run_linux_untar
+
+
+@runs_on([['distributed-replicated', 'replicated'], ['glusterfs']])
+class TestCompileLinuxKernelWithSelfHeal(GlusterBaseClass):
+
+    def setUp(self):
+
+        self.get_super_method(self, 'setUp')()
+
+        self.first_client = self.mounts[0].client_system
+        self.mountpoint = self.mounts[0].mountpoint
+
+        # Setup Volume
+        if not self.setup_volume_and_mount_volume([self.mounts[0]]):
+            raise ExecutionError("Failed to setup and mount volume")
+
+    def tearDown(self):
+
+        if not self.unmount_volume_and_cleanup_volume([self.mounts[0]]):
+            raise ExecutionError("Failed to cleanup Volume")
+
+        # Calling GlusterBaseClass tearDown
+        self.get_super_method(self, 'tearDown')()
+
+    def _untar_linux_kernel_in_a_specific_dir(self):
+        """A function to create files and dirs on mount point"""
+        # Create a parent directory test_link_self_heal on mount point
+        ret = mkdir(self.first_client,
+                    '{}/{}'.format(self.mountpoint, 'test_self_heal'))
+        self.assertTrue(ret, "Failed to create dir test_self_heal")
+
+        # Start linux untar on dir linuxuntar
+        proc = run_linux_untar(self.clients[0], self.mounts[0].mountpoint,
+                               dirs=tuple(['test_self_heal']))[0]
+        try:
+            ret, _, _ = proc.async_communicate()
+            if not ret:
+                untar_done = False
+            untar_done = True
+        except ValueError:
+            untar_done = True
+        self.assertTrue(untar_done,
+                        "Kernel untar not done on client mount point")
+
+    def _bring_bricks_offline(self):
```

Comment:
This private method is used only once. So, you can keep the logic inside the test method

Reply:
The intention of keeping it a private method is to reuse the same file for multiple similar cases. Another advantage to having it is that I don't what to disable pylint checks instead this helps us not hit them at all. Nacking this request as I feel it's invalid.

Reply:
Are you sure, that we have more MRTC cases relevant to this and will use this private function?